integration_test.go

```diff
@@ -263,6 +264,44 @@ func TestIntegrationProtocols(t *testing.T) {
 	}
 }
 
+func TestIntegrationThriftEnvelope(t *testing.T) {
+	tests := []struct {
+		disableEnvelope    bool
+		wantEnvelopeHeader string
+	}{
+		{true, "false"},
+		{false, "true"},
+	}
+
+	for _, tt := range tests {
+
+		var called bool
+		server := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+			called = true
+			envelope := r.Header.Get(transport.HTTPThriftEnvelopeheader)
+			assert.Equal(t, tt.wantEnvelopeHeader, envelope)
+		}))
+		defer server.Close()
```

Comment:
nit: Because this isn't a function, all the servers will keep running until the test exits. I would push this into an anonymous function.